Fabulous.XamarinForms/src/Fabulous.XamarinForms.Core/ViewUpdaters.fs

```diff
@@ -28,6 +28,120 @@ module ViewUpdaters =
+    ///
+    /// The algorithm will try in priority to update elements sharing the same instance (usually achieved with dependsOn)
+    /// or sharing the same key. All other elements will try to reuse existing previous elements where possible.
+    /// If no reuse is possible, the element will create a new control.
+    let updateCollectionGenericInternal
+            (prevCollOpt: 'T[] voption)
+            (collOpt: 'T[] voption)
+            (getKey: 'T -> string voption)
+            (canReuse: 'T -> 'T -> bool)
+            (move: int -> 'T -> unit)
+            (create: int -> 'T -> unit)
+            (update: int -> 'T -> 'T -> unit)
+            (remove: int -> unit)
+            (clear: unit -> unit) =
+        
+        match prevCollOpt, collOpt with 
+        | ValueSome prevColl, ValueSome newColl when identical prevColl newColl -> ()
+        | ValueSome prevColl, ValueSome newColl when prevColl <> null && newColl <> null && prevColl.Length = 0 && newColl.Length = 0 -> ()
+        | _, ValueNone -> clear ()
+        | _, ValueSome coll when (coll = null || coll.Length = 0) -> clear ()
+        | _, ValueSome coll ->
+            
+            let prevColl =
+                match prevCollOpt with
+                | ValueSome x -> x
+                | _ -> [||]
+
+            // Count the existing targetColl
+            let n = prevColl.Length
+            
+            let availableKeyedElements = Dictionary<string,'T>()
+            let rest= ResizeArray<'T>()
+            let newKeys = HashSet<string>()
+            
+            for i in 0 .. coll.Length-1 do
+                 let newChild = coll.[i]
+                 let key = getKey newChild
+                 match key with
+                 | ValueSome key -> newKeys.Add key |> ignore
+                 | ValueNone _ -> ()
+                 
+            for i in 0..n-1 do
+                let key = getKey prevColl.[i]
+                match key with
+                | ValueSome key ->
+                    availableKeyedElements.[key] <- prevColl.[i]
+                | ValueNone -> rest.Add prevColl.[i]
+            
+            let contains (key:string voption) =
+                match key with
+                | ValueSome key -> availableKeyedElements.ContainsKey key
+                | ValueNone -> false
+                
+            let newKeyContains el =
+                match getKey el with
+                | ValueSome key -> newKeys.Contains key
+                | ValueNone -> false
+  
+            // Adjust the existing targetColl and create the new targetColl
+            for i in 0 .. coll.Length-1 do
+                let newChild = coll.[i]
+                let key = getKey newChild
+                let elems =
+                        seq{
+                            yield! rest
+                            yield! availableKeyedElements.Values
+                           }
+                if (prevColl|>Seq.exists (identical newChild)) then
```

Comment:
Nitpick: `prevColl` is known to be an array, so I think it's clearer to use `Array.exists` instead of `Seq.exists` :D

Reply:
Ok